src/Symfony/Component/Routing/Loader/Configurator/Traits/AddTrait.php

```diff
@@ -24,22 +25,66 @@
 
     private $name = '';
 
+    private $prefixes;
+
     /**
      * Adds a route.
+     *
+     * @param string|array $path the path, or the localized paths of the route
      */
-    final public function add(string $name, string $path): RouteConfigurator
+    final public function add(string $name, $path): RouteConfigurator
```

Comment:
IMO it's better to offer a separate method like `addLocalized` where we can keep type declarations and makes the intention clear

Reply:
do you not agree?

Reply:
sounds good to me

Reply:
I prefer the current way, because each new method adds to the autocompletion list and amplifies the choice complexity.

Reply:
Then the best method is `doIt(...$args)` no more method needed

Reply:
@Tobion please take the community participation guidelines into account. There's a world of difference for having one param with mixed argument handling and that. No need to be sarcastic about it.

Reply:
I think one of the strong points of the PHP routing configurator is discoverability. Invoking the autocompletion, I can see by the method name and arguments type and name how to use the method and the purpose. With this change, we are removing that ability. I now have to read the documentation, see examples or inspect the implementation to know the mixed accepted values and their effect.
The autocompletion list in this scope is not large at all and we could worry about that in the future if that really becomes an issue.

Reply:
@andersonamuller did you actually try it on your IDE? AFAIK, the `@param` in the phpdoc provides exactly what your are describing.

Reply:
@nicolas-grekas I did, I use PHPStorm. I either have to have a new tab view open with the method documentation or select the method and the look for the documentation, but in the autocomplete hover list it does not show anything from the PHPDoc part.

Reply:
This is basically what it looks like in PhpStorm by default:

<img width="830" alt="bildschirmfoto 2018-03-21 um 17 13 07" src="https://user-images.githubusercontent.com/1957048/37722178-a1e7cdf2-2d2b-11e8-8e7d-b33253ad66ab.png">

Reply:
@frankdejonge I don't think I deserve being lectured by you.
1. My comment was not directed at you at all. So I cannot have hurt your feelings which is what the guidelines tries to prevent.
2. Of the thousands review comments I did, how many of them did not respect the guidelines in you're opinion that I deserve being disciplined?
3. I hope Nicolas does not feel disrespected by me considering how much praise I gave him for all his work so far.

Reply:
@Tobion your comment was sarcastic for sure :) But I didn't take it personally so all is fine on my side. Still nice to take care of each other.

@xabbuh thanks for the screenshot. On my side, all is good as is.